test/html.test.js

```diff
@@ -1602,6 +1607,47 @@ QUnit.test('gopsSafeToAlignWith returns correct list', function() {
     'empty array when no gops in buffer come after currentTime');
 });
 
+QUnit.test('currentGOPStart returns time of most recent GOP', function(assert) {
+  const pts = (time) => Math.ceil(time * 90000);
+  let mapping = 0;
+  let currentTime = 0;
+  let buffer = [];
+  let player;
+  let actual;
+  let expected = null;
+
+  actual = currentGOPStart(buffer, player, mapping);
+  QUnit.deepEqual(actual, expected, 'null when player is undefined');
+
+  player = { currentTime: () => currentTime };
+  actual = currentGOPStart(buffer, player, mapping);
+  QUnit.deepEqual(actual, expected, 'null when buffer is empty');
+
+  buffer = [
+    { pts: pts(1) },
+    { pts: pts(2.2) },
+    { pts: pts(3) }
+  ];
+  actual = currentGOPStart(buffer, player, mapping);
+  QUnit.deepEqual(actual, expected, 'null when entire buffer is ahead of currentTime');
+
+  currentTime = 1.5;
+  actual = currentGOPStart(buffer, player, mapping);
+  QUnit.deepEqual(actual, 1, 'uses previous GOP when between GOP starts');
+
+  currentTime = 2.2;
+  actual = currentGOPStart(buffer, player, mapping);
+  QUnit.deepEqual(actual, 2.200011111111111, 'uses currentTime when currentTime === GOP start');
```

Comment:
this assertion threw me off as I didn't notice `pts` was using the ceiling of the time input at first